,dep,version,filepath,bloated,cve,affected versions,advisory
0,usd-core,<=23.5,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,no one,no one,no one
1,tqdm,>=4.51.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,no one,no one,no one
2,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2014-3589,"<2.3.2,>=2.5.0,<2.5.2","Pillow versions 2.3.2 and 2.5.2 include a fix for CVE-2014-3589: PIL/IcnsImagePlugin.py in Python Imaging Library (PIL) and Pillow before 2.3.2 and 2.5.x before 2.5.2 allows remote attackers to cause a denial of service via a crafted block size.
https://github.com/python-pillow/Pillow/commit/205e056f8f9b06ed7b925cf8aa0874bc4aaf8a7d"
3,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2020-15999,<8.0.1,Pillow 8.0.1 updates 'FreeType' used in binary wheels to v2.10.4 to include a security fix.
4,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2020-35654,<8.1.0,Pillow 8.1.0 fixes TIFF OOB Write error. CVE-2020-35654 #5175.
5,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2020-35653,<8.1.0,"In Pillow before 8.1.0, PcxDecode has a buffer over-read when decoding a crafted PCX file because the user-supplied stride value is trusted for buffer calculations."
6,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2020-35655,<8.1.0,Pillow 8.1.0 includes a fix for SGI Decode buffer overrun. CVE-2020-35655 #5173.
7,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-25289,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25289: TiffDecode has a heap-based buffer overflow when decoding crafted YCbCr files because of certain interpretation conflicts with LibTIFF in RGBA mode. NOTE: this issue exists because of an incomplete fix for CVE-2020-35654.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
8,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-25290,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25290: In TiffDecode.c, there is a negative-offset memcpy with an invalid size.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
9,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-27922,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-27922: Pillow before 8.1.1 allows attackers to cause a denial of service (memory consumption) because the reported size of a contained image is not properly checked for an ICNS container, and thus an attempted memory allocation can be very large.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
10,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-25291,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25291: In TiffDecode.c, there is an out-of-bounds read in TiffreadRGBATile via invalid tile boundaries.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
11,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-27921,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-27921: Pillow before 8.1.1 allows attackers to cause a denial of service (memory consumption) because the reported size of a contained image is not properly checked for a BLP container, and thus an attempted memory allocation can be very large.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
12,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-25293,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25293: There is an out-of-bounds read in SGIRleDecode.c.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
13,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-25292,<8.1.1,"Pillow 8.1.1 includes a fix for CVE-2021-25292: The PDF parser allows a regular expression DoS (ReDoS) attack via a crafted PDF file because of a catastrophic backtracking regex.
https://pillow.readthedocs.io/en/stable/releasenotes/8.1.1.html"
14,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-25288,<8.2.0,"Pillow 8.2.0 includes a fix for CVE-2021-25288: There is an out-of-bounds read in J2kDecode, in j2ku_gray_i.
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-25287-cve-2021-25288-fix-oob-read-in-jpeg2kdecode"
15,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-28677,<8.2.0,"Pillow version 8.2.0 includes a fix for CVE-2021-28677: For EPS data, the readline implementation used in EPSImageFile has to deal with any combination of \r and \n as line endings. It used an accidentally quadratic method of accumulating lines while looking for a line ending. A malicious EPS file could use this to perform a DoS of Pillow in the open phase, before an image was accepted for opening.
https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MQHA5HAIBOYI3R6HDWCLAGFTIQP767FL/
https://github.com/python-pillow/Pillow/pull/5377
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-28677-fix-eps-dos-on-open"
16,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-28678,<8.2.0,"Pillow version 8.2.0 includes a fix for CVE-2021-28678: For BLP data, BlpImagePlugin did not properly check that reads (after jumping to file offsets) returned data. This could lead to a DoS where the decoder could be run a large number of times on empty data.
https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MQHA5HAIBOYI3R6HDWCLAGFTIQP767FL/
https://github.com/python-pillow/Pillow/pull/5377
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-28678-fix-blp-dos"
17,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-28676,<8.2.0,"Pillow version 8.2.0 includes a fix for CVE-2021-28676: For FLI data, FliDecode did not properly check that the block advance was non-zero, potentially leading to an infinite loop on load.
https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MQHA5HAIBOYI3R6HDWCLAGFTIQP767FL/
https://github.com/python-pillow/Pillow/pull/5377
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-28676-fix-fli-dos"
18,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-25287,<8.2.0,"Pillow 8.2.0 includes a fix for CVE-2021-25287: There is an out-of-bounds read in J2kDecode, in j2ku_graya_la.
https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#cve-2021-25287-cve-2021-25288-fix-oob-read-in-jpeg2kdecode"
19,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-34552,<8.3.0,"Pillow 8.3.0 includes a fix for CVE-2021-34552: Pillow through 8.2.0 and PIL (also known as Python Imaging Library) through 1.1.7 allow an attacker to pass controlled parameters directly into a convert function to trigger a buffer overflow in Convert.c
https://pillow.readthedocs.io/en/stable/releasenotes/8.3.0.html#buffer-overflow
https://pillow.readthedocs.io/en/stable/releasenotes/index.html"
20,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2022-22816,<9.0.0,"Pillow 9.0.0 includes a fix for CVE-2022-22816: path_getbbox in path.c in Pillow before 9.0.0 has a buffer over-read during initialization of ImagePath.Path.
https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#fixed-imagepath-path-array-handling"
21,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,PVE-2021-44525,<9.0.0,"Pillow 9.0.0 excludes carriage return in PDF regex to help prevent ReDoS.
https://github.com/python-pillow/Pillow/pull/5912
https://github.com/python-pillow/Pillow/commit/43b800d933c996226e4d7df00c33fcbe46d97363"
22,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,PVE-2022-44524,<9.0.0,"Pillow 9.0.0 ensures JpegImagePlugin stops at the end of a truncated file to avoid Denial of Service attacks.
https://github.com/python-pillow/Pillow/pull/5921
https://github.com/advisories/GHSA-4fx9-vc88-q2xc"
23,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2022-22815,<9.0.0,"Pillow 9.0.0 includes a fix for CVE-2022-22815: path_getbbox in path.c in Pillow before 9.0.0 improperly initializes ImagePath.Path.
https://pillow.readthedocs.io/en/stable/releasenotes/9.0.0.html#fixed-imagepath-path-array-handling"
24,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2022-24303,<9.0.1,Pillow before 9.0.1 allows attackers to delete files because spaces in temporary pathnames are mishandled.
25,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2022-22817,<9.0.1,"Pillow 9.0.1 includes a fix for CVE-2022-22817: PIL.ImageMath.eval in Pillow before 9.0.0 allows evaluation of arbitrary expressions, such as ones that use the Python exec method. A first patch was issued for version 9.0.0 but it did not prevent builtins available to lambda expressions.
https://pillow.readthedocs.io/en/stable/releasenotes/9.0.1.html#security"
26,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2022-45198,<9.2.0,Pillow before 9.2.0 performs Improper Handling of Highly Compressed GIF Data (Data Amplification).
27,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-28675,">=0,<8.2.0",An issue was discovered in Pillow before 8.2.0. PSDImagePlugin.PsdImageFile lacked a sanity check on the number of input layers relative to the size of the data block. This could lead to a DoS on Image.open prior to Image.load.
28,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-27923,">=4.3.0,<8.1.1","Pillow before 8.1.1 allows attackers to cause a denial of service (memory consumption) because the reported size of a contained image is not properly checked for an ICO container, and thus an attempted memory allocation can be very large."
29,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-23437,">=5.2.0,<8.3.2","Pillow from 5.2.0 and before 8.3.2 is vulnerable to Regular Expression Denial of Service (ReDoS) via the getrgb function.
https://github.com/python-pillow/Pillow/commit/9e08eb8f78fdfd2f476e1b20b7cf38683754866b
https://pillow.readthedocs.io/en/stable/releasenotes/8.3.2.html"
30,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2022-30595,">=9.1.0,<9.1.1",libImaging/TgaRleDecode.c in Pillow 9.1.0 has a heap buffer overflow in the processing of invalid TGA image files.
31,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,PVE-2023-55182,">=9.1.0,<9.3.0","Pillow 9.3.0 includes a security fix: Pillow will now decode the data in its natural CMYK mode, then convert it to RGB and rearrange the channels afterwards. Trying to load the data in an incorrect mode could result in a segmentation fault.
https://pillow.readthedocs.io/en/stable/releasenotes/9.3.0.html#decode-jpeg-compressed-blp1-data-in-original-mode"
32,Pillow,>=8.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2022-45199,">=9.2.0,<9.3.0","Pillow before 9.3.0 allows denial of service via SAMPLESPERPIXEL.
https://pillow.readthedocs.io/en/stable/releasenotes/9.3.0.html#limit-samplesperpixel-to-avoid-runtime-dos"
33,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2017-12852,<1.13.2,"Numpy 1.13.2 includes a fix for CVE-2017-12852: The numpy.pad function in Numpy 1.13.1 and older versions is missing input validation. An empty list or ndarray will stick into an infinite loop, which can allow attackers to cause a DoS attack.
https://github.com/numpy/numpy/issues/9560#issuecomment-322395292"
34,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2019-6446,<1.16.3,"Numpy 1.16.3 includes a fix for CVE-2019-6446: It uses the pickle Python module unsafely, which allows remote attackers to execute arbitrary code via a crafted serialized object, as demonstrated by a numpy.load call.
https://github.com/numpy/numpy/commit/89b688732b37616c9d26623f81aaee1703c30ffb"
35,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-33430,<1.21.0rc1,"Numpy 1.21.0rc1 includes a fix for CVE-2021-33430: A Buffer Overflow vulnerability in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service. 
NOTE: The vendor does not agree this is a vulnerability. In (very limited) circumstances a user may be able provoke the buffer overflow, the user is most likely already privileged to at least provoke denial of service by exhausting memory. Triggering this further requires the use of uncommon API (complicated structured dtypes), which is very unlikely to be available to an unprivileged user.
https://github.com/numpy/numpy/issues/18939"
36,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-34141,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-34141: An incomplete string comparison in the numpy.core component in NumPy before 1.22.0 allows attackers to trigger slightly incorrect copying by constructing specific string objects. 
NOTE: the vendor states that this reported code behavior is ""completely harmless.""
https://github.com/numpy/numpy/issues/18993"
37,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-41496,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-41496: Buffer overflow in the array_from_pyobj function of fortranobject.c, which allows attackers to conduct a Denial of Service attacks by carefully constructing an array with negative values. 
NOTE: The vendor does not agree this is a vulnerability; the negative dimensions can only be created by an already privileged user (or internally).
https://github.com/numpy/numpy/issues/19000"
38,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2021-41495,<1.22.2,"Numpy 1.22.2  includes a fix for CVE-2021-41495: Null Pointer Dereference vulnerability exists in numpy.sort in NumPy in the PyArray_DescrNew function due to missing return-value validation, which allows attackers to conduct DoS attacks by repetitively creating sort arrays. 
NOTE: While correct that validation is missing, an error can only occur due to an exhaustion of memory. If the user can exhaust memory, they are already privileged. Further, it should be practically impossible to construct an attack which can target the memory exhaustion to occur at exactly this place.
https://github.com/numpy/numpy/issues/19038"
39,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2014-1859,<1.8.1,"Numpy 1.8.1 includes a fix for CVE-2014-1859: (1) core/tests/test_memmap.py, (2) core/tests/test_multiarray.py, (3) f2py/f2py2e.py, and (4) lib/tests/test_io.py in NumPy before 1.8.1 allow local users to write to arbitrary files via a symlink attack on a temporary file.
https://github.com/numpy/numpy/pull/4262"
40,numpy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2014-1858,<1.8.1,"Numpy 1.8.1 includes a fix for CVE-2014-1858: __init__.py in f2py in NumPy before 1.8.1 allows local users to write to arbitrary files via a symlink attack on a temporary file.
https://github.com/numpy/numpy/commit/0bb46c1448b0d3f5453d5182a17ea7ac5854ee15"
41,scipy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,CVE-2013-4251,">=0,<0.12.1",The scipy.weave component in SciPy before 0.12.1 creates insecure temporary directories.
42,pybind11,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/requirements.txt,False,no one,no one,no one
43,ipython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2015-4707,<3.2.0,Cross-site scripting (XSS) vulnerability in IPython before 3.2 allows remote attackers to inject arbitrary web script or HTML via vectors involving JSON error messages and the /api/notebooks path.
44,ipython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2015-5607,<3.2.1,"Ipython 3.2.1 includes a fix for CVE-2015-5607: Cross-site request forgery in the REST API in IPython 2 and 3.
https://github.com/ipython/ipython/commit/1415a9710407e7c14900531813c15ba6165f0816"
45,ipython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2015-6938,<3.2.2,"Cross-site scripting (XSS) vulnerability in the file browser in notebook/notebookapp.py in IPython Notebook before 3.2.2 and Jupyter Notebook 4.0.x before 4.0.5 allows remote attackers to inject arbitrary web script or HTML via a folder name.  NOTE: this was originally reported as a cross-site request forgery (CSRF) vulnerability, but this may be inaccurate."
46,ipython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2015-7337,<3.2.2,"The editor in IPython Notebook before 3.2.2 and Jupyter Notebook 4.0.x before 4.0.5 allows remote attackers to execute arbitrary JavaScript code via a crafted file, which triggers a redirect to files/, related to MIME types."
47,ipython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2023-24816,<8.10.0,"IPython 8.10.0 includes a fix for CVE-2023-24816: Versions prior to 8.10.0 are subject to a command injection vulnerability with very specific prerequisites. This vulnerability requires that the function 'IPython.utils.terminal.set_term_title' be called on Windows in a Python environment where ctypes is not available. The dependency on 'ctypes' in 'IPython.utils._process_win32' prevents the vulnerable code from ever being reached in the ipython binary. However, as a library that could be used by another tool 'set_term_title' could be called and hence introduce a vulnerability. If an attacker get untrusted input to an instance of this function they would be able to inject shell commands as current process and limited to the scope of the current process. As a workaround, users should ensure that any calls to the 'IPython.utils.terminal.set_term_title' function are done with trusted or filtered input.
https://github.com/ipython/ipython/security/advisories/GHSA-29gw-9793-fvw7"
48,ipython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2014-3429,">=0.12,<1.2","IPython Notebook 0.12 through 1.x before 1.2 does not validate the origin of websocket requests, which allows remote attackers to execute arbitrary code by leveraging knowledge of the kernel id and a crafted page."
49,ipython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2016-6524,">=2.0.0,<4.0.0",Ipython 4.0 includes a fix for CVE-2016-6524: Untrusted LaTeX output could be added to the page in a way that could execute javascript.
50,ipython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2015-4706,">=3.0.0,<3.2.0",Cross-site scripting (XSS) vulnerability in IPython 3.x before 3.2 allows remote attackers to inject arbitrary web script or HTML via vectors involving JSON error messages and the /api/contents path.
51,ipython,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2022-21699,">=8.0.0a0,<8.0.1,>=7.17.0,<7.31.1,>=6.0.0a0,<7.16.3,<5.11","Ipython versions 8.0.1, 7.31.1, 7.16.3 and 5.11 include a fix for CVE-2022-21699: Affected versions are subject to an arbitrary code execution vulnerability achieved by not properly managing cross user temporary files. This vulnerability allows one user to run code as another on the same machine.
https://github.com/ipython/ipython/security/advisories/GHSA-pq7m-3gw7-gq5x"
52,ipython,< 8.13,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2014-3429,">=0.12,<1.2","IPython Notebook 0.12 through 1.x before 1.2 does not validate the origin of websocket requests, which allows remote attackers to execute arbitrary code by leveraging knowledge of the kernel id and a crafted page."
53,ipython,< 8.13,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2016-6524,">=2.0.0,<4.0.0",Ipython 4.0 includes a fix for CVE-2016-6524: Untrusted LaTeX output could be added to the page in a way that could execute javascript.
54,ipython,< 8.13,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2015-4706,">=3.0.0,<3.2.0",Cross-site scripting (XSS) vulnerability in IPython 3.x before 3.2 allows remote attackers to inject arbitrary web script or HTML via vectors involving JSON error messages and the /api/contents path.
55,ipython,< 8.13,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2022-21699,">=8.0.0a0,<8.0.1,>=7.17.0,<7.31.1,>=6.0.0a0,<7.16.3,<5.11","Ipython versions 8.0.1, 7.31.1, 7.16.3 and 5.11 include a fix for CVE-2022-21699: Affected versions are subject to an arbitrary code execution vulnerability achieved by not properly managing cross user temporary files. This vulnerability allows one user to run code as another on the same machine.
https://github.com/ipython/ipython/security/advisories/GHSA-pq7m-3gw7-gq5x"
56,ipycanvas,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,no one,no one,no one
57,pyzmq,<25,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,no one,no one,no one
58,tornado,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2012-2374,<2.2.1,CRLF injection vulnerability in the tornado.web.RequestHandler.set_header function in Tornado before 2.2.1 allows remote attackers to inject arbitrary HTTP headers and conduct HTTP response splitting attacks via crafted input.
59,tornado,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2023-28370,<6.3.2,Tornado 6.3.2 includes a fix for CVE-2023-28370: Open redirect vulnerability in Tornado versions 6.3.1 and earlier allows a remote unauthenticated attacker to redirect a user to an arbitrary web site and conduct a phishing attack by having user access a specially crafted URL.
60,tornado,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2014-9720,">=0,<3.2.2","Tornado before 3.2.2 sends arbitrary responses that contain a fixed CSRF token and may be sent with HTTP compression, which makes it easier for remote attackers to conduct a BREACH attack and determine this token via a series of crafted requests."
61,ipyevents,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,no one,no one,no one
62,flask,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2019-1010083,<0.12.3,"Flask 0.12.3 includes a fix for CVE-2019-1010083: Unexpected memory usage. The impact is denial of service. The attack vector is crafted encoded JSON data. NOTE: this may overlap CVE-2018-1000656.
https://github.com/pallets/flask/pull/2695/commits/0e1e9a04aaf29ab78f721cfc79ac2a691f6e3929"
63,flask,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2018-1000656,<0.12.3,flask version Before 0.12.3 contains a CWE-20: Improper Input Validation vulnerability in flask that can result in Large amount of memory usage possibly leading to denial of service. This attack appear to be exploitable via Attacker provides JSON data in incorrect encoding. This vulnerability appears to have been fixed in 0.12.3.
64,flask,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,PVE-2021-25820,<0.6.1,flask 0.6.1 fixes a security problem that allowed clients to download arbitrary files  if the host server was a windows based operating system and the client  uses backslashes to escape the directory the files where exposed from.
65,flask,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,CVE-2023-30861,"<2.2.5,>=2.3.0,<2.3.2","Flask is a lightweight WSGI web application framework. When all of the following conditions are met, a response containing data intended for one client may be cached and subsequently sent by the proxy to other clients. If the proxy also caches 'Set-Cookie' headers, it may send one client's 'session' cookie to other clients. The severity depends on the application's use of the session and the proxy's behavior regarding cookies. The risk depends on all these conditions being met.

1. The application must be hosted behind a caching proxy that does not strip cookies or ignore responses with cookies.
2. The application sets 'session.permanent = True'
3. The application does not access or modify the session at any point during a request.
4. 'SESSION_REFRESH_EACH_REQUEST' enabled (the default).
5. The application does not set a 'Cache-Control' header to indicate that a page is private or should not be cached.

This happens because vulnerable versions of Flask only set the 'Vary: Cookie' header when the session is accessed or modified, not when it is refreshed (re-sent to update the expiration) without being accessed or modified. This issue has been fixed in versions 2.3.2 and 2.2.5."
66,jupyter_client,<8,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,no one,no one,no one
67,comm,>=0.1.3,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/viz_requirements.txt,False,no one,no one,no one
68,flake8-comprehensions,==3.2.2,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/ci_requirements.txt,False,no one,no one,no one
69,flake8,==3.7.9,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/ci_requirements.txt,False,no one,no one,no one
70,flake8-bugbear,==20.1.4,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/ci_requirements.txt,False,no one,no one,no one
71,flake8-pyi,==19.3.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/ci_requirements.txt,False,no one,no one,no one
72,flake8-mypy,==17.8.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/ci_requirements.txt,False,no one,no one,no one
73,nbmake,==1.4.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/ci_requirements.txt,False,no one,no one,no one
74,pytest-cov,==3.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/ci_requirements.txt,False,no one,no one,no one
75,pytest,==7.1.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/ci_requirements.txt,False,no one,no one,no one
76,jupyter,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/ci_requirements.txt,False,no one,no one,no one
77,setuptools,==58.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,CVE-2022-40897,<65.5.1,"Python Packaging Authority (PyPA) setuptools before 65.5.1 allows remote attackers to cause a denial of service via HTML in a crafted package or custom PackageIndex page. There is a Regular Expression Denial of Service (ReDoS) in package_index.py.
https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages"
78,sphinx_rtd_theme,==1.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,no one,no one,no one
79,sphinx,">=3.5.4,<5.1.0",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,no one,no one,no one
80,urllib3,"<1.27,>=1.21.1",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,CVE-2018-20060,<1.23,"urllib3 before version 1.23 does not remove the Authorization HTTP header when following a cross-origin redirect (i.e., a redirect that differs in host, port, or scheme). This can allow for credentials in the Authorization header to be exposed to unintended hosts or transmitted in cleartext."
81,urllib3,"<1.27,>=1.21.1",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,CVE-2019-11324,<1.24.2,"The urllib3 library before 1.24.2 for Python mishandles certain cases where the desired set of CA certificates is different from the OS store of CA certificates, which results in SSL connections succeeding in situations where a verification failure is the correct outcome. This is related to use of the ssl_context, ca_certs, or ca_certs_dir argument."
82,urllib3,"<1.27,>=1.21.1",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,CVE-2020-26137,<1.25.9,"Urllib3 1.25.9 includes a fix for CVE-2020-26137: Urllib3 before 1.25.9 allows CRLF injection if the attacker controls the HTTP request method, as demonstrated by inserting CR and LF control characters in the first argument of putrequest(). NOTE: this is similar to CVE-2020-26116.
https://github.com/python/cpython/issues/83784"
83,urllib3,"<1.27,>=1.21.1",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,CVE-2021-33503,<1.26.5,"Urllib3 1.26.5 includes a fix for CVE-2021-33503: When provided with a URL containing many @ characters in the authority component, the authority regular expression exhibits catastrophic backtracking, causing a denial of service if a URL were passed as a parameter or redirected to via an HTTP redirect."
84,urllib3,"<1.27,>=1.21.1",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,CVE-2019-11236,<=1.24.1,"In the urllib3 library through 1.24.1 for Python, CRLF injection is possible if the attacker controls the request parameter."
85,urllib3,"<1.27,>=1.21.1",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,CVE-2020-7212,">=1.25.2,<=1.25.7","The _encode_invalid_chars function in util/url.py in the urllib3 library 1.25.2 through 1.25.7 for Python allows a denial of service (CPU consumption) because of an inefficient algorithm. The percent_encodings array contains all matches of percent encodings. It is not deduplicated. For a URL of length N, the size of percent_encodings may be up to O(N). The next step (normalize existing percent-encoded bytes) also takes up to O(N) for each step, so the total time is O(N^2). If percent_encodings were deduplicated, the time to compute _encode_invalid_chars would be O(kN), where k is at most 484 ((10+6*2)^2). See: CVE-2020-7212."
86,urllib3,"<1.27,>=1.21.1",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/doc_requirements.txt,False,CVE-2021-28363,">=1.26.0,<1.26.4","Urllib3 1.26.4 includes a fix for CVE-2021-28363: The urllib3 library 1.26.x before 1.26.4 for Python omits SSL certificate validation in some cases involving HTTPS to HTTPS proxies. The initial connection to the HTTPS proxy (if an SSLContext isn't given via proxy_config) doesn't verify the hostname of the certificate. This means certificates for different servers that still validate properly with the default urllib3 SSLContext will be silently accepted.
https://github.com/urllib3/urllib3/security/advisories/GHSA-5phf-pp7p-vc2r"
87,cython,==0.29.20,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/tools/build_requirements.txt,False,no one,no one,no one
88,numpy,"<1.27.0,>=1.19.5",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/docs/readthedocs_requirements.txt,False,CVE-2021-33430,<1.21.0rc1,"Numpy 1.21.0rc1 includes a fix for CVE-2021-33430: A Buffer Overflow vulnerability in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service. 
NOTE: The vendor does not agree this is a vulnerability. In (very limited) circumstances a user may be able provoke the buffer overflow, the user is most likely already privileged to at least provoke denial of service by exhausting memory. Triggering this further requires the use of uncommon API (complicated structured dtypes), which is very unlikely to be available to an unprivileged user.
https://github.com/numpy/numpy/issues/18939"
89,numpy,"<1.27.0,>=1.19.5",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/docs/readthedocs_requirements.txt,False,CVE-2021-34141,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-34141: An incomplete string comparison in the numpy.core component in NumPy before 1.22.0 allows attackers to trigger slightly incorrect copying by constructing specific string objects. 
NOTE: the vendor states that this reported code behavior is ""completely harmless.""
https://github.com/numpy/numpy/issues/18993"
90,numpy,"<1.27.0,>=1.19.5",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/docs/readthedocs_requirements.txt,False,CVE-2021-41496,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-41496: Buffer overflow in the array_from_pyobj function of fortranobject.c, which allows attackers to conduct a Denial of Service attacks by carefully constructing an array with negative values. 
NOTE: The vendor does not agree this is a vulnerability; the negative dimensions can only be created by an already privileged user (or internally).
https://github.com/numpy/numpy/issues/19000"
91,numpy,"<1.27.0,>=1.19.5",/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/docs/readthedocs_requirements.txt,False,CVE-2021-41495,<1.22.2,"Numpy 1.22.2  includes a fix for CVE-2021-41495: Null Pointer Dereference vulnerability exists in numpy.sort in NumPy in the PyArray_DescrNew function due to missing return-value validation, which allows attackers to conduct DoS attacks by repetitively creating sort arrays. 
NOTE: While correct that validation is missing, an error can only occur due to an exhaustion of memory. If the user can exhaust memory, they are already privileged. Further, it should be practically impossible to construct an attack which can target the memory exhaustion to occur at exactly this place.
https://github.com/numpy/numpy/issues/19038"
92,scipy,==1.10.1,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/docs/readthedocs_requirements.txt,False,CVE-2013-4251,">=0,<0.12.1",The scipy.weave component in SciPy before 0.12.1 creates insecure temporary directories.
93,torch,==1.8.2+cpu,/home/daniele/git/NICHE_projects/Repo/Cloneables/kaolin/docs/readthedocs_requirements.txt,False,CVE-2022-45907,">=0,<1.13.1","Torch 1.13.1 includes a fix for CVE-2022-45907: In PyTorch before trunk/89695, torch.jit.annotations.parse_type_line can cause arbitrary code execution because eval is used unsafely.
https://github.com/pytorch/pytorch/issues/88868"
